### PR TITLE
Fix bugs

### PR DESCRIPTION
**Pull request by suargi**
Thank you for making your code available.

I have found some bugs:
- https://github.com/AntreasAntoniou/HowToTrainYourMAMLPytorch/issues/24)
- https://github.com/AntreasAntoniou/HowToTrainYourMAMLPytorch/issues/42
- Hyperparameters specified in configurations are not the same ones used in the original MAML.
- Update code to latest version of pytorch (where to place scheduler and how to properly retrieve the last lr)
- Bug when adding images to the datasets (first image for each label was always duplicated)
- Bug when normalizing the dataset: `dataset is datasets_dict["train"]` was never true.

It would be interesting to replicate the study and compare again against the original MAML results.

### CODE REVIEW

**Comment by suargi** [comment]
Fixed some more bugs with training/validation.
Some functions had hard-coded training mode, thus when using validation and test datasets, we were still computing gradients and not using statistics properly.

For now, I reverted the configuration files, so I can easily detect whether I break something
